internal/workflow/localact/save_preprocessing_tasks.go

```diff
@@ -39,16 +40,22 @@ func SavePreprocessingTasksActivity(
 	params SavePreprocessingTasksActivityParams,
 ) (*SavePreprocessingTasksActivityResult, error) {
 	var res SavePreprocessingTasksActivityResult
-	for _, t := range params.Tasks {
-		task := preprocTaskToTask(t)
-		task.WorkflowUUID = params.WorkflowUUID
-		// TODO: Create deterministic UUIDs and make activities idempotent.
-		task.UUID = uuid.Must(uuid.NewRandomFromReader(params.RNG))
-
-		if err := params.Ingestsvc.CreateTask(ctx, &task); err != nil {
-			return &res, fmt.Errorf("SavePreprocessingTasksActivity: %v", err)
+	seq := persistence.TaskSequence(func(yield func(*datatypes.Task) bool) {
+		for _, t := range params.Tasks {
```

Comment:
There doesn't seem to be any advantage in this case to using an iterator, as we already have the slice of `[]*datatypes.Task` in memory.  I understand that an iterator is useful in other situations and it doesn't make sense to have a different `CreateTasks()` method that takes a slice, I'm just noting that I don't think it's a benefit here.

Reply:
We start from `[]preprocessing.Task`, so the iterator lets us map each entry to a `*datatype.Task` and stream it without first building a `[]*datatypes.Task`. That's perhaps the small usability gain from the perspective of the producer if anything.